src/main/java/org/jabref/gui/preferences/customentrytypes/CustomEntryTypesTab.java

```diff
@@ -143,6 +147,26 @@ private void setupEntryTypesTable() {
 
     private void setupFieldsTable() {
         fieldNameColumn.setCellValueFactory(item -> item.getValue().nameProperty());
+        fieldNameColumn.setCellFactory(TextFieldTableCell.forTableColumn());
+        fieldNameColumn.setEditable(true);
+        fieldNameColumn.setOnEditCommit(
+                (TableColumn.CellEditEvent<FieldViewModel, String> event) -> {
+                    String newFieldValue = StringUtil.capitalizeFirst(event.getNewValue());
+                    UnknownField field = (UnknownField) event.getRowValue().getField();
+                    EntryTypeViewModel selectedEntryType = viewModel.selectedEntryTypeProperty().get();
+                    ObservableList<FieldViewModel> entryFields = selectedEntryType.fields();
+                    boolean fieldExists = entryFields.stream().anyMatch(fieldViewModel ->
+                            fieldViewModel.nameProperty().getValue().equalsIgnoreCase(newFieldValue));
+
+                    if (!fieldExists) {
+                        event.getRowValue().setField(newFieldValue);
+                        field.setName(newFieldValue);
+                    } else {
+                        dialogService.notify(Localization.lang("Duplicate fields: You added field \"%0\" twice. Only one will be kept.", newFieldValue));
+                        event.getTableView().edit(-1, null);
+                        event.getTableView().refresh();
+                    }
```

Comment:
Can you swat these conditions? Both branches are similarly long - and negations are sometimes hard to read. `if (fieldExists)` reads more natural